src/main/java/com/zendesk/maxwell/MaxwellConfig.java

```diff
@@ -57,31 +66,89 @@ public void setInitialPosition(BinlogPosition position) throws IOException {
 		}
 	}
 
-	public static MaxwellConfig fromPropfile(String filename) throws IOException {
+	private void parseOptions(String [] argv) {
+		OptionParser parser = new OptionParser();
+		parser.accepts( "host" ).withRequiredArg();
+		parser.accepts( "password" ).withRequiredArg();
+		parser.accepts( "user" ).withRequiredArg();
+		parser.accepts( "port" ).withRequiredArg();
+		parser.accepts( "producer" ).withRequiredArg();
+
+		OptionSet options = parser.parse(argv);
+
+		if ( options.has("host"))
+			this.mysqlHost = (String) options.valueOf("host");
+		if ( options.has("password"))
+			this.mysqlPassword = (String) options.valueOf("password");
+		if ( options.has("user"))
+			this.mysqlUser = (String) options.valueOf("user");
+		if ( options.has("port"))
+			this.mysqlPort = Integer.valueOf((String) options.valueOf("port"));
+		if ( options.has("producer"))
+			this.producer = (String) options.valueOf("producer");
```

Comment:
Maybe rename producer -> producerType. We also have a getProducer() in here which actually creates the producer object.